projeto completo (#1)
betrybe/sd-0x-project-pixels-art-rubric: README.md

    - [10 - Faça o quadro de pixels ter seu tamanho definido pelo usuário.](#10---faça-o-quadro-de-pixels-ter-seu-tamanho-definido-pelo-usuário)
    - [11 - Limite o tamanho do mínimo e máximo do board.](#11---limite-o-tamanho-do-mínimo-e-máximo-do-board)
    - [12 - Faça com que as cores da paleta sejam geradas aleatoriamente ao carregar a página.](#12---faça-com-que-as-cores-da-paleta-sejam-geradas-aleatoriamente-ao-carregar-a-página)
- [Depois de terminar o desenvolvimento](#depois-de-terminar-o-desenvolvimento)
- [Revisando um pull request](#revisando-um-pull-request)
- [Avisos Finais](#avisos-finais)

---

# Habilidades

- Manipular o DOM

- Manipular o Javascript

---

## O que deverá ser desenvolvido

- Você irá implementar uma página web que contém uma paleta de cores funcional que poderá ser utilizada para criar desenhos em pixels. Para isto você irá utilizar `javascript`, `css` e `html`

---

### Desenvolvimento

- Implemente uma paleta de cores usando `javascript`, `css` e `html`.

---

## Data de Entrega

  - Serão `X` dias de projeto.
  - Data de entrega para avaliação final do projeto: `DD/MM/YYYY - 14:00h`.

---

## Instruções para entregar seu projeto:

> ### 🗒ANTES DE COMEÇAR A DESENVOLVER:

Reviewer: Faltou a instrução de rodar o `npm install` aqui. Acham válido?

1. Clone o repositório

>   * `git clone https://github.com/tryber/sd-0x-project-pixels-art.git`.

Reviewer: O que acha de alterar o padrão do método de autenticação para ssh?

Author: entao precisa saber pq todos os projetos estao com https , pq so os de fundamentos tem que estar ssh?

    - [11 - Limite o tamanho do mínimo e máximo do board.](#11---limite-o-tamanho-do-mínimo-e-máximo-do-board)
    - [12 - Faça com que as cores da paleta sejam geradas aleatoriamente ao carregar a página.](#12---faça-com-que-as-cores-da-paleta-sejam-geradas-aleatoriamente-ao-carregar-a-página)
- [Depois de terminar o desenvolvimento](#depois-de-terminar-o-desenvolvimento)
- [Revisando um pull request](#revisando-um-pull-request)
- [Avisos Finais](#avisos-finais)

---

# Habilidades

- Manipular o DOM

- Manipular o Javascript

---

# Entregáveis

Para entregar o seu projeto você deverá criar um Pull Request neste repositório.

Este Pull Request deverá conter os arquivos `index.html`, `style.css` e `script.js`, que conterão seu código HTML, CSS e JavaScript, respectivamente.

## ⚠️ É importante que seus arquivos tenham exatamente estes nomes! ⚠️

Você pode adicionar outros arquivos se julgar necessário. Qualquer dúvida, procure a monitoria.

Lembre-se que você pode consultar nosso conteúdo sobre [Git & GitHub](https://course.betrybe.com/intro/git/) sempre que precisar!

---

## O que deverá ser desenvolvido

> - Você irá implementar uma paleta de cores usando `javascript` e `css`.

Reviewer 2: ```suggestion
- Você irá implementar uma página web que contém uma paleta de cores funcional que poderá ser utilizada para criar desenhos em pixels. Para isto você irá utilizar `javascript`, `css` e `html`.
```

- [Depois de terminar o desenvolvimento](#depois-de-terminar-o-desenvolvimento)
- [Revisando um pull request](#revisando-um-pull-request)
- [Avisos Finais](#avisos-finais)

---

# Habilidades

- Manipular o DOM

- Manipular o Javascript

---

## O que deverá ser desenvolvido

- Você irá implementar uma página web que contém uma paleta de cores funcional que poderá ser utilizada para criar desenhos em pixels. Para isto você irá utilizar `javascript`, `css` e `html`

---

### Desenvolvimento

- Implemente uma paleta de cores usando `javascript`, `css` e `html`.

---

## Data de Entrega

  - Serão `X` dias de projeto.
  - Data de entrega para avaliação final do projeto: `DD/MM/YYYY - 14:00h`.

---

## Instruções para entregar seu projeto:

### 🗒ANTES DE COMEÇAR A DESENVOLVER:

1. Clone o repositório
  * `git clone git@github.com:tryber/sd-0x-project-pixels-art.git`.
  * Entre na pasta do repositório que você acabou de clonar:
    * `cd sd-0x-project-pixels-art`

2. Instale as dependências e inicialize o projeto
  * Instale as dependências:
    * `npm install`

>   * Inicialize o projeto:
>     * `npm start` (uma nova página deve abrir no seu navegador com um texto simples)

Reviewer 3: Como nos outros projetos, esse script não está listado no package.json. Nesses projetos iniciais usamos a extensão Live Server do VSCode. Acredito que essas linhas tenham sido adicionadas por engano, faz sentido?

```suggestion
```

  * Usando o exemplo anterior: `git push -u origin mariazinha-meme-generator-project`

6. Crie um novo `Pull Request` _(PR)_
  * Vá até a página de _Pull Requests_ do [repositório no GitHub](https://github.com/tryber/sd-0x-project-pixels-art/pulls)
  * Clique no botão verde _"New pull request"_
  * Clique na caixa de seleção _"Compare"_ e escolha a sua branch **com atenção**
  * Clique no botão verde _"Create pull request"_

>   * Adicione uma descrição para o _Pull Request_, um título claro que o identifique, e clique no botão verde _"Create pull request"_

Reviewer: Eu acredito que essa instrução do jeito que está da muita margem pra pessoa estudante colocar qualquer título no PR. O que você acha de sermos mais diretos ainda? 

```suggestion
  * Adicione uma descrição para o _Pull Request_, um título que o identifique, e clique no botão verde _"Create pull request"_. Crie da seguinte forma: `[MARIAZINHA] Projeto Pixel Art`.
```

Author: entao isso é apenas um guia, não é nada obrigatorio, mesmo tendo esse guia pessoal faz os titulos nada parecido  com isso , nao acho que vale a pena prender estudante a isso

  * **Não se preocupe em preencher mais nada por enquanto!**
  * Volte até a [página de _Pull Requests_ do repositório](https://github.com/tryber/sd-0x-project-pixels-art/pulls) e confira que o seu _Pull Request_ está criado

---

## Durante o desenvolvimento

> * ⚠ **PULL REQUESTS COM ISSUES NO LINTER NÃO SERÃO AVALIADAS, ATENTE-SE PARA RESOLVÊ-LAS ANTES DE FINALIZAR O DESENVOLVIMENTO!** ⚠

Reviewer: Nesse projeto o Linter não é obrigatório. Sendo assim, o que acha de remover qualquer referência de obrigação do linter nesse readme do projeto?

- Se precisar consultar os valores do _CSS_ de um elemento a partir do _JavaScript_, [dê uma olhada aqui](https://www.w3schools.com/jsref/jsref_getcomputedstyle.asp).

- Para alterar alguma propriedade do _CSS_ de um elemento através do _JavaScript_, dê uma olhada no [atributo `style`](https://www.w3schools.com/jsref/prop_style_backgroundcolor.asp) do elemento.

- Para colocar sua página no [GitHub Pages](https://pages.github.com/), não é necessário remover o conteúdo que já está lá, você pode apenas adicionar essa nova página. Para isso, todo o conteúdo desse projeto deve ser colocado em uma pasta `/projetos/pixels-art`.

## Linter

Para garantir a qualidade do código, vamos utilizar neste projeto os linters `ESLint` e `StyleLint`.
Assim o código estará alinhado com as boas práticas de desenvolvimento, sendo mais legível
e de fácil manutenção! Para rodá-los localmente no projeto, execute os comandos abaixo:

  ```bash

> npm run lint

Reviewer: O que acha de explicar no readme a diferença entre o `run lint` e o `run lint:style`?

npm run lint:styles
```

> ⚠️ **PULL REQUESTS COM ISSUES DE LINTER NÃO SERÃO AVALIADAS.

Reviewer: O que acha de substituir essa mensagem por uma orientando que o linter não é obrigatório nesse projeto? Pelo que eu vi, existe uma mensagem padrão que foi aplicada no projeto Lessons Learned. :P

  * **Não se preocupe em preencher mais nada por enquanto!**
  * Volte até a [página de _Pull Requests_ do repositório](https://github.com/tryber/sd-0x-project-pixels-art/pulls) e confira que o seu _Pull Request_ está criado

---

## Durante o desenvolvimento

* ⚠ **PULL REQUESTS COM ISSUES NO LINTER NÃO SERÃO AVALIADAS, ATENTE-SE PARA RESOLVÊ-LAS ANTES DE FINALIZAR O DESENVOLVIMENTO!** ⚠

* Faça `commits` das alterações que você fizer no código regularmente

* Lembre-se de sempre após um (ou alguns) `commits` atualizar o repositório remoto

* Os comandos que você utilizará com mais frequência são:
  1. `git status` _(para verificar o que está em vermelho - fora do stage - e o que está em verde - no stage)_
  2. `git add` _(para adicionar arquivos ao stage do Git)_
  3. `git commit` _(para criar um commit com os arquivos que estão no stage do Git)_
  4. `git push` _(para enviar o commit para o repositório remoto após o passo anterior)_
  5. `git push -u nome-da-branch` _(para enviar o commit para o repositório remoto na primeira vez que fizer o `push` de uma nova branch)_
  
---

> # Como desenvolver

Reviewer 2: Acredito que neste tópico **Como desenvolver** pode ser:

```
Este Pull Request deverá conter os arquivos `index.html`, `style.css` e `script.js`, que conterão seu código HTML, CSS e JavaScript, respectivamente.

## ⚠️ É importante que seus arquivos tenham exatamente estes nomes! ⚠️

Você pode adicionar outros arquivos se julgar necessário. Qualquer dúvida, procure a monitoria.
```

Este trecho está nos **Entregáveis** acho que pode sair de lá.

- A paleta de cores não deve conter cores repetidas.

**O que será verificado:**

- A paleta de cores deve ser um elemento com `id` denominado `color-palette`

> - Verifica se cada cor individual da paleta de cores deve possuir a `classe` `color`.

Reviewer 2: ```suggestion
- Verifica se cada cor individual da paleta de cores possui a `classe` `color`.
```

> - Verifica se a cor de fundo de cada elemento da paleta deverá ser a cor que o elemento representa. **A única cor não permitida na paleta é a cor branca.**

Reviewer 2: ```suggestion
- Verifica se a cor de fundo de cada elemento da paleta é a cor que o elemento representa. **A única cor não permitida na paleta é a cor branca.**
```

> - Verifica se cada elemento da paleta de cores deverá ter uma borda preta, sólida e com 1 pixel de largura;

Reviewer 2: ```suggestion
- Verifica se cada elemento da paleta de cores tem uma borda preta, sólida e com 1 pixel de largura;
```

> - Verifica se a paleta deverá listar todas as cores disponíveis para utilização, lado a lado

Reviewer 2: ```suggestion
- Verifica se a paleta lista todas as cores disponíveis para utilização, lado a lado.
```

> - Verifica se a paleta de cores deverá ser posicionada abaixo do título \'Paleta de Cores\'

Reviewer 2: ```suggestion
- Verifica se a paleta de cores está posicionada abaixo do título \'Paleta de Cores\'
```

> - Verifica se a paleta de cores não deve conter cores repetidas.

Reviewer 2: ```suggestion
- Verifica se a paleta de cores não contém cores repetidas.
```

- A paleta de cores não deve conter cores repetidas.

**O que será verificado:**

- A paleta de cores deve ser um elemento com `id` denominado `color-palette`

> - Verifica se cada cor individual da paleta de cores possui a `classe` `color`.

Reviewer: Faz sentido sermos mais nítidos na comunicação?

```suggestion
- Verifica se cada cor individual da paleta de cores possui a `classe` chamada `color`.
```

-  O quadro de "pixels" deve aparecer abaixo da paleta de cores.

**O que será verificado:**

> - Verifica se o quadro de pixels deve possuir o `id` denominado `pixel-board`

Reviewer 2: ```suggestion
- Verifica se o quadro de pixels possui o `id` denominado `pixel-board`
```

> - Verifica se cada pixel individual dentro do quadro deve possuir a `classe` denominada `pixel`.

Reviewer 2: ```suggestion
- Verifica se cada pixel individual dentro do quadro possui a `classe` denominada `pixel`.
```

> - Verifica se a cor inicial dos pixels dentro do quadro, ao abrir a página, deve ser branca.

Reviewer 2: ```suggestion
- Verifica se a cor inicial dos pixels dentro do quadro, ao abrir a página, é branca.
```

> - Verifica se o quadro de pixels deve aparecer abaixo da paleta de cores

Reviewer 2: ```suggestion
- Verifica se o quadro de pixels aparece abaixo da paleta de cores
```

### 5 - Aplique a cada elemento do quadro de pixels, deve possuir 40 pixels de largura e 40 pixels de altura e ser delimitado por uma borda preta de 1 pixel.

**O que será verificado:**

- Verifica se o quadro de pixels tem altura e comprimento de 5 elementos

> - Verifica se 40 pixels deve ser o tamanho total do elemento, incluindo seu conteúdo e excluindo a borda preta, que deve ser criada à parte.

Reviewer 2: ```suggestion
- Verifica se 40 pixels é o tamanho total do elemento, incluindo seu conteúdo e excluindo a borda preta, que deve ser criada à parte.
```

### 6 - Executar o carregamento da página, a cor preta da paleta já deve estar selecionada para pintar os pixels.

- O elemento da cor preta deve possuir, inicialmente, a `classe` `selected`;

- Note que o elemento que deverá receber a classe `selected` deve ser um dos elementos que possuem a classe `color`, como especificado no **requisito 2**.

**O que será verificado:**

> - Verifica se o elemento da cor preta deve possuir, inicialmente, a `classe` `selected`

Reviewer 2: ```suggestion
- Verifica se o elemento da cor preta possui, inicialmente, a `classe` `selected`
```

- Verifica se nenhuma outra cor da paleta tem a `classe` `selected`

### 7 - Clicar em uma das cores da paleta, a cor selecionada é que vai ser usada para preencher os pixels no quadro.

- A `classe` `selected` deve ser adicionada à cor selecionada na paleta, ao mesmo tempo em que é removida da cor anteriormente selecionada;

- Somente uma das cores da paleta deve ter a `classe` `selected` de cada vez;

- Note que os elementos que deverão receber a classe `selected` devem ser os mesmos elementos que possuem a classe `color`, como especificado no **requisito 2**.

**O que será verificado:**

- Verifica se somente uma cor da paleta de cores tem a classe `selected` de cada vez

> - Verifica se os pixels dentro do quadro não podem ter a classe `selected` quando são clicados

Reviewer 2: ```suggestion
- Verifica se os pixels dentro do quadro não têm a classe `selected` quando são clicados
```

### 8 - Clicar em um pixel dentro do quadro após selecionar uma cor na paleta, o pixel deve ser preenchido com esta cor.

**O que será verificado:**

- Verifica se ao carregar a página deve ser possível pintar os pixels de preto

> - Verifica se após selecionar uma outra cor na paleta, deve ser possível pintar os pixels com essa cor

Reviewer 2: ```suggestion
- Verifica se após selecionar uma outra cor na paleta, é possível pintar os pixels com essa cor
```

> - Verifica se somente o pixel que foi clicado deverá ser preenchido com a cor selecionada, sem influenciar na cor dos demais pixels.

Reviewer 2: ```suggestion
- Verifica se somente o pixel que foi clicado foi preenchido com a cor selecionada, sem influenciar na cor dos demais pixels.
```

### 9 - Crie um botão que, ao ser clicado, limpa o quadro preenchendo a cor de todos seus pixels com branco.

**O que será verificado:**

> - Verifica se o botão deve ter o `id` denominado `clear-board`

Reviewer 2: ```suggestion
- Verifica se o botão tem o `id` denominado `clear-board`
```

> - Verifica se o botão deve estar posicionado entre a paleta de cores e o quadro de pixels

Reviewer 2: ```suggestion
- Verifica se o botão está posicionado entre a paleta de cores e o quadro de pixels
```

> - Verifica se o texto do botão deve ser \'Limpar\'

Reviewer 2: ```suggestion
- Verifica se o texto do botão é \'Limpar\'
```

- O novo quadro deve ter todos os pixels preenchidos com a cor branca.

**O que será verificado:**

- Verifica se existe um input com o id `board-size`

- Verifica se existe um botão com o id `generate-board`

> - Verifica se o input só deve aceitar número maiores que zero. Essa restrição deve ser feita usando os atributos do elemento `input`

Reviewer 2: ```suggestion
- Verifica se o input só aceita número maiores que zero. Essa restrição deve ser feita usando os atributos do elemento `input`
```

> - Verifica se o botão deve conter o texto \'VQV\'

Reviewer 2: ```suggestion
- Verifica se o botão contém o texto \'VQV\'
```

> - Verifica se o input deve estar posicionado entre a paleta de cores e o quadro de pixels

Reviewer 2: ```suggestion
- Verifica se o input está posicionado entre a paleta de cores e o quadro de pixels
```

> - Verifica se o botão deve estar posicionado ao lado do input

Reviewer 2: ```suggestion
- Verifica se o botão está posicionado ao lado do input
```

> - Verifica se nenhum valor for colocado no input ao clicar no botão, mostre um `alert` com o texto: \'Board inválido!\'

Reviewer 2: ```suggestion
- Verifica se nenhum valor for colocado no input ao clicar no botão, um `alert` é exibido com o texto: \'Board inválido!\'
```

- Verifica se ao clicar no botão com um valor preenchido, o tamanho do board muda.

> - Verifica se o novo quadro deve ter todos os pixels preenchidos com a cor branca

Reviewer 2: ```suggestion
- Verifica se o novo quadro tem todos os pixels preenchidos com a cor branca
```

### 11 - Limite o tamanho do mínimo e máximo do board.

- Caso o valor digitado no input `board-sze` fuja do intervalo de 5 a 50, faça:

  - Valor menor que 5, considerar 5 como padrão;

  - Valor maior que 50, considerar 50 como padrão.

**O que será verificado:**

- Verifica se a altura máxima do board é 50

- Verifica se a altura do board é 5 quando um valor menor é colocado no input

- Verifica se a altura do board é 50 quando um valor maior é colocado no input

### 12 - Faça com que as cores da paleta sejam geradas aleatoriamente ao carregar a página.

- A cor preta ainda precisa estar presente e deve ser a primeira na sua paleta de cores.

**O que será verificado:**

- Verifica se as cores geradas na paleta são diferentes a cada carregamento da página

> - Verifica se a cor preta ainda precisa estar presente e deve ser a primeira na sua paleta de cores

Reviewer 2: ```suggestion
- Verifica se a cor preta ainda está presente e é a primeira na sua paleta de cores
```

---

> ## Dicas

Reviewer 2: Creio que o tópico **Dicas** está fora do local correto. Acredito que deva estar presente como um sub-tópico do tópico "**Como desenvolver**"
